eq-author/src/components/ContentPicker/ContentPicker.js

```diff
@@ -18,6 +18,65 @@ const ActionButtons = styled(ButtonGroup)`
   padding: 1em 0;
   flex: 0 0 auto;
 `;
+
+const getDesiredPath = (
+  dataset,
+  selectedId,
+  config,
+  currentIndex,
+  path = []
+) => {
+  let dataIndex;
+  for (
+    dataIndex = 0;
+    dataIndex < dataset[config[currentIndex].childKey].length;
+    dataIndex++
+  ) {
+    if (config[currentIndex + 1].childKey) {
+      const result = getDesiredPath(
+        dataset[config[currentIndex].childKey][dataIndex],
+        selectedId,
+        config,
+        currentIndex + 1,
+        [...path, { [config[currentIndex].childKey]: dataIndex }]
+      );
+      if (result) {
+        return result;
+      }
+    } else if (
+      dataset[config[currentIndex].childKey][dataIndex].id === selectedId
+    ) {
+      return [...path, { [config[currentIndex].childKey]: dataIndex }];
+    }
+  }
+};
+
+const setSelectedElement = (config, selectedId, data) => {
```

Comment:
This looks like it could be extracted to a separate file which would make it easy to test independently.